utils/predicate/predicate_bytes.go

```diff
@@ -66,3 +67,20 @@ func BytesToHashSlice(b []byte) []common.Hash {
 	}
 	return hashes
 }
+
+// GetPredicateResultBytes returns the predicate result bytes from the extra data.
+// If the extra data does not contain predicate result bytes, an error is returned.
+func GetPredicateResultBytes(extraData []byte) ([]byte, error) {
+	// Prior to the DUpgrade, the VM enforces the extra data is smaller than or equal to this size.
+	// After the DUpgrade, the VM pre-verifies the extra data past the dynamic fee rollup window is
+	// valid.
+	if len(extraData) < params.DynamicFeeExtraDataSize {
+		return nil, fmt.Errorf("header extra data too short for predicate verification found: %d, required: %d", len(extraData), params.DynamicFeeExtraDataSize)
+	}
+	return extraData[params.DynamicFeeExtraDataSize:], nil
+}
+
+// HasPredicateResultBytes returns true if the extra data contains predicate result bytes.
+func HasPredicateResultBytes(extraData []byte) bool {
```

Comment:
seems unused? not against adding this as an exported function for later use though

Reply:
let's remove it for now.